unit_test/CMakeLists.txt

```diff
@@ -39,19 +39,21 @@ IF (KOKKOS_ENABLE_CUDA)
   KOKKOSKERNELS_INCLUDE_DIRECTORIES(REQUIRED_DURING_INSTALLATION_TESTING ${CMAKE_CURRENT_SOURCE_DIR}/cuda)
 
   APPEND_GLOB(CUDA_BLAS_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/cuda/Test_Cuda_Blas*.cpp)
-  KOKKOSKERNELS_ADD_UNIT_TEST_EXECUTABLE_AND_TEST(
+  KOKKOSKERNELS_ADD_UNIT_TEST(
```

Comment:
I see from above that you refactored the code a bit to only use the new `KOKKOSKERNELS_ADD_UNIT_TEST` and now we add a new field `COMPONENT` to each test so that they get compiled only `KOKKOSKERNELS_IS_ENABLED` returns true for that particular component.